src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs

```diff
@@ -97,78 +98,85 @@ internal static async Task<bool> IsOverrideAsync(Solution solution, ISymbol memb
+                                if (nameToLookFor == null)
+                                    continue;
 
-                            if (interfaceType.MemberNames.Contains(nameToLookFor))
-                            {
-                                foreach (var m in interfaceType.GetMembers(symbol.Name))
+                                if (interfaceType.MemberNames.Contains(nameToLookFor))
                                 {
-                                    var sourceMethod = await FindSourceDefinitionAsync(m, solution, cancellationToken).ConfigureAwait(false);
-                                    var bestMethod = sourceMethod ?? m;
-
-                                    var implementations = await type.FindImplementationsForInterfaceMemberAsync(bestMethod, solution, cancellationToken).ConfigureAwait(false);
-                                    foreach (var implementation in implementations)
+                                    foreach (var m in interfaceType.GetMembers(symbol.Name))
                                     {
-                                        if (implementation != null &&
-                                            SymbolEquivalenceComparer.Instance.Equals(implementation.OriginalDefinition, symbol.OriginalDefinition))
+                                        var sourceMethod = await FindSourceDefinitionAsync(m, solution, cancellationToken).ConfigureAwait(false);
+                                        var bestMethod = sourceMethod ?? m;
+
+                                        var implementations = await type.FindImplementationsForInterfaceMemberAsync(bestMethod, solution, cancellationToken).ConfigureAwait(false);
+                                        foreach (var implementation in implementations)
                                         {
-                                            builder.Add(bestMethod);
+                                            if (implementation != null &&
+                                                SymbolEquivalenceComparer.Instance.Equals(implementation.OriginalDefinition, symbol.OriginalDefinition))
+                                            {
+                                                builder.Add(bestMethod);
+                                            }
                                         }
                                     }
                                 }
                             }
                         }
-                    }
 
-                    return builder.Distinct(SymbolEquivalenceComparer.Instance).ToImmutableArray();
+                        return builder.Distinct(SymbolEquivalenceComparer.Instance).ToImmutableArray();
+                    }
                 }
-            }
 
-            return ImmutableArray<ISymbol>.Empty;
+                return ImmutableArray<ISymbol>.Empty;
+            }
+            catch (Exception ex) when (FatalError.ReportAndPropagateUnlessCanceled(ex, cancellationToken, ErrorSeverity.Diagnostic))
```

Comment:
It's often helpful to put a comment here just linking to the bug so we know what this is for.

Reply:
done.